# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/VinayBhavikatti/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Refresh / Content Opportunity Scoring

I want to find which content pages should be reviewed first for a possible refresh, improvement, protection, or monitoring. This lane fits the data because it contains search performance signals such as impressions, clicks, and average position. I want to use these signals to create an evidence-based ranking that helps a reviewer decide which pages deserve attention first.

In [9]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

My question is which content pages should be reviewed first for a possible refresh or improvement. The decision is which pages deserve attention first, and a content/SEO reviewer would act on the recommendation. A wrong recommendation could waste time and resources on a page that does not need improvement, while a missed page could lose an opportunity to improve its search performance.

In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

url = "https://raw.githubusercontent.com/VinayBhavikatti/flyrank-ml-internship/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Rows:", len(df))
print("Columns:", list(df.columns))

print("\nUseful numbers for my lane:")
print("Total pages:", len(df))
print("Average impressions (90d):", round(df["impressions_90d"].mean(), 2))
print("Average clicks (90d):", round(df["clicks_90d"].mean(), 2))
print("Average position:", round(df["avg_position"].mean(), 2))

Rows: 30000
Columns: ['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']

Useful numbers for my lane:
Total pages: 30000
Average impressions (90d): 5200.37
Average clicks (90d): 16.1
Average position: 16.34


The starter dataset contains 30,000 content pages. On average, pages received 5200.37 impressions and 16.11 clicks over 90 days, with an average position of 16.34. These numbers show that the dataset contains measurable search performance differences that can be used to identify which pages may deserve review first.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

I can make observed and measured statements about search performance in the dataset, and use the signals to provide directional recommendations about which pages may deserve attention first. The analysis can support decisions about content review and improvement. However, I cannot claim that the observed relationships are causal, prove that a change will produce a specific result, or predict Google rankings with certainty.

In [12]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.